pocs/utils/google/storage.py

```diff
@@ -97,3 +99,247 @@ def upload(self, local_path, remote_path=None):
+
+        return output_path
+
+    def lookup_fits_header(self, remote_blob, parse_line=None):
+        """Read the FITS header from storage.
+
+        FITS Header Units are stored in blocks of 2880 bytes consisting of 36 lines
+        that are 80 bytes long each. The Header Unit always ends with the single
+        word 'END' on a line (not necessarily line 36).
+
+        Here the header is streamed from Storage until the 'END' is found, with
+        each line given minimal parsing.
+
+        See https://fits.gsfc.nasa.gov/fits_primer.html for overview of FITS format.
+
+        Args:
+            remote_blob (google.cloud.storage.blob.Blob): Blob object
+                corresponding to stored FITS file.
+            parse_line (callable, optional): A callable function which will be
+                passed an 80 character string corresponding to each line in the header.
+
+        Returns:
+            dict: FITS header as a dictonary.
+        """
+        i = 1
+        if remote_blob.name.endswith('.fz'):
+            i = 2  # We skip the compression header info
+
+        headers = dict()
+
+        streaming = True
+        while streaming:
+            # Get a header card
+            start_byte = 2880 * (i - 1)
+            end_byte = (2880 * i) - 1
+            b_string = remote_blob.download_as_string(start=start_byte, end=end_byte)
+
+            # Loop over 80-char lines
+            for j in range(0, len(b_string), 80):
+                item_string = b_string[j:j + 80].decode()
+
+                # End of FITS Header, stop streaming
+                if item_string.startswith('END'):
+                    streaming = False
+                    break
+
+                # If custom parse function supplied, call that
+                if parse_line is not None:
+                    try:
+                        parse_line(item_string)
+                    except TypeError:
+                        warn('parse_line must be callable')
+                    continue
+
+                # Get key=value pairs (skip COMMENTS and HISTORY)
+                if item_string.find('=') > 0:
+                    k, v = item_string.split('=')
+
+                    # Remove FITS comment
+                    if ' / ' in v:
+                        v = v.split(' / ')[0]
+
+                    v = v.strip()
+
+                    # Cleanup and discover type in dumb fashion
+                    if v.startswith("'") and v.endswith("'"):
+                        v = v.replace("'", "").strip()
+                    elif v.find('.') > 0:
+                        v = float(v)
+                    elif v == 'T':
+                        v = True
+                    elif v == 'F':
+                        v = False
+                    else:
+                        v = int(v)
+
+                    headers[k.strip()] = v
+
+            i += 1
+
+        return headers
+
+
+def upload_observation_to_bucket(pan_id,
```

Comment:
How about merging most of the body of this into the class above at some point? A TODO will be fine.